src/lib.rs

```diff
@@ -446,7 +446,13 @@ pub async fn get_nwc_events(
             .await?
             .unwrap_or_default();
         for nwc_event in nwc_events {
-            events.push(nwc_event);
+            events.push(nwc_event.clone());
+            // now delete responses since we don't care after sending
+            if kind == Kind::WalletConnectResponse {
+                if let Err(e) = delete_nwc_response(nwc_event, ctx).await {
+                    console_log!("failed to delete nwc response: {e}");
+                }
+            }
```

Comment:
if you make `delete_nwc_response` take a reference to the event, you can do the delete first then do `events.push()` without the clone

Reply:
Fixed